cdap-client/src/main/java/co/cask/cdap/client/NamespaceClient.java

```diff
@@ -0,0 +1,144 @@
+  }
+
+  /**
+   * * Deletes a namespace from CDAP.
+   *
+   * @param namespaceId id of the namespace to be deleted.
+   * @throws IOException if a network error occurred
+   * @throws UnAuthorizedAccessTokenException if the request is not authorized successfully in the gateway server
+   * @throws NotFoundException if the specified namespace is not found
+   * @throws CannotBeDeletedException if the specified namespace is reserved and cannot be deleted
+   */
+  public void delete(String namespaceId) throws IOException, UnAuthorizedAccessTokenException, NotFoundException,
+    CannotBeDeletedException {
+    HttpResponse response = restClient.execute(HttpMethod.DELETE,
+                                               config.resolveURL(String.format("namespaces/%s", namespaceId)),
+                                               config.getAccessToken(),
+                                               HttpURLConnection.HTTP_NOT_FOUND,
+                                               HttpURLConnection.HTTP_FORBIDDEN);
+    if (HttpURLConnection.HTTP_NOT_FOUND == response.getResponseCode()) {
+      throw new NotFoundException(NAMESPACE_ENTITY_TYPE, namespaceId);
+    }
+    if (HttpURLConnection.HTTP_FORBIDDEN == response.getResponseCode()) {
+      throw new CannotBeDeletedException(NAMESPACE_ENTITY_TYPE, namespaceId);
+    }
+  }
+
+  /**
+   * Creates a new namespace in CDAP
+   *
+   * @param namespaceMeta the {@link NamespaceMeta} for the namespace to be created
+   * @throws IOException if a network error occurred
+   * @throws UnAuthorizedAccessTokenException if the request is not authorized successfully in the gateway server
+   * @throws AlreadyExistsException if the specified namespace already exists
+   * @throws BadRequestException if the specified namespace contains an invalid or reserved namespace id
+   */
+  public void create(NamespaceMeta namespaceMeta)
+    throws IOException, UnAuthorizedAccessTokenException, AlreadyExistsException, BadRequestException {
+    URL url = config.resolveURL(String.format("namespaces/%s", namespaceMeta.getId()));
+    NamespaceMeta.Builder builder = new NamespaceMeta.Builder();
+    String displayName = namespaceMeta.getDisplayName();
+    String description = namespaceMeta.getDescription();
+    if (displayName != null) {
+      builder.setDisplayName(displayName);
+    }
+    if (description != null) {
+      builder.setDescription(description);
+    }
+    String body = new Gson().toJson(builder.build());
+    HttpRequest request = HttpRequest.put(url).withBody(body).build();
+    HttpResponse response = restClient.upload(request, config.getAccessToken(), HttpURLConnection.HTTP_BAD_REQUEST);
+    String responseBody = response.getResponseBodyAsString();
+    if (response.getResponseCode() == HttpURLConnection.HTTP_BAD_REQUEST) {
+      throw new BadRequestException("Bad request: " + responseBody);
+    }
+    if (responseBody != null && responseBody.contains("already exists")) {
```

Comment:
There is no better way other than string matching unfortunately. The create API is a PUT API. So we wanted to make it idempotent. Hence even if the namespace already exists, it returns a 200 with a message that the namespace already exists in the response body. 

However, I agree that we can make this check a little stricter. Have added an ``equals`` check as opposed to the current ``contains`` check. Let me know what you think.